spec/expression.dd

```diff
@@ -2006,6 +2006,43 @@ $(H3 $(LEGACY_LNAME2 StringLiteral, string_literals, String Literals))
         C style implicit concatenation without an intervening operator is
         error prone and not supported in D.)
 
+
+$(H3 $(LNAME2 hex_string_literals, Hex String Literals))
+    $(P Because hex string literals contain binary data not limited to textual data, they allow additional conversions over other string literals.)
+
+    $(P A hex string literal implicitly converts to a constant `byte[]` or `ubyte[]`.)
+        $(SPEC_RUNNABLE_EXAMPLE_RUN
+        -------------
+        immutable ubyte[] b = x"3F 80 00 00";
+        const byte[] c = x"3F 80 00 00";
+        -------------
+        )
+
+    $(P A hex string literal can be explicitly cast to an array of integers with a larger size than 1.
+        A big endian byte order in the hex string will be assumed.)
+
+        $(SPEC_RUNNABLE_EXAMPLE_RUN
+        -------------
+        immutable uint[] data = cast(immutable uint[]) x"AABBCCDD";
```

Comment:
../dmd/generated/linux/release/64/../../../../druntime/import/core/internal/array/casting.d(96): Error: reinterpreting cast from `string` to `Array*` is not supported in CTFE

Reply:
I didn't realize the other dmd pull is not merged, nevermind